packages/mom5/package.py

```diff
@@ -312,33 +470,48 @@ def build(self, spec, prefix):
         # ./MOM_compile.csh --type $mom_type --platform spack
         with working_dir(join_path(self.stage.source_path, "exp")):
             build = Executable("./MOM_compile.csh")
-
-            if "+optimisation_report" in self.spec:
-                build.add_default_env("REPORT", "true")
-            if "+restart_repro" in self.spec:
-                build.add_default_env("REPRO", "true")
-
-            # The MOM5 commit d7ba13a3f364ce130b6ad0ba813f01832cada7a2
-            # requires the --no_version switch to avoid git hashes being
-            # embedded in the binary.
-            build(
-                "--type",
-                self.spec.variants["type"].value,
-                "--platform",
-                self._platform,
-                "--no_environ",
-                "--no_version"
-            )
+            if "@access-esm1.5" in self.spec:
+                if "+restart_repro" in self.spec:
+                    build.add_default_env("REPRO", "true")
+                build(
+                    "--type",
+                    "ACCESS-CM",
+                    "--platform",
+                    self._platform,
+                    "--no_environ"
+                )
+            else:
+                if "+optimisation_report" in self.spec:
+                    build.add_default_env("REPORT", "true")
+                if "+restart_repro" in self.spec:
+                    build.add_default_env("REPRO", "true")
+
+                # The MOM5 commit d7ba13a3f364ce130b6ad0ba813f01832cada7a2
+                # requires the --no_version switch to avoid git hashes being
+                # embedded in the binary.
+                build(
+                    "--type",
+                    self.spec.variants["type"].value,
+                    "--platform",
+                    self._platform,
+                    "--no_environ",
+                    "--no_version"
+                )
 
     def install(self, spec, prefix):
 
         mkdirp(prefix.bin)
+        if "@access-esm1.5" in self.spec:
+            type_value = "ACCESS-CM"
```

Comment:
Can you set `self.spec.variants["type"].value = "ACCESS-CM"` much earlier if the version is access-esm1.5.

Reply:
Done, except that I had to use "esm-type". Spack takes the last `variant` call that uses the same name, only keeping `when` from previous `variant` calls. See https://github.com/spack/spack/blob/develop/lib/spack/docs/packaging_guide.rst#overriding-variants